src/dmd/hdrgen.d

```diff
@@ -2265,6 +2265,48 @@ public:
         expToBuffer(e.e2, cast(PREC)(precedence[e.op] + 1), buf, hgs);
     }
 
+    override void visit(CommaExp e)
+    {
+        // CommaExp is generated by the compiler so it shouldn't
+        // appear in error messages or header files.
+        // For now, this treats the case where the compiler
+        // generates CommaExp for temporaries by calling
+        // the `sideeffect.copyToTemp` function.
+        auto ve = e.e2.isVarExp();
+
+        // not a CommaExp introduced for temporaries, go on
+        // the old path
+        if (!ve || !ve.var.storage_class & STC.temp)
+        {
+            visit(cast(BinExp)e);
+            return;
+        }
+
+        // CommaExp that contain temporaries inserted via
+        // `copyToTemp` are usually of the form
+        // ((T __temp = exp), __tmp).
+        // Asserts are here to easily spot
+        // missing cases where CommaExp
+        // are used for other constructs
+        auto vd = ve.var.isVarDeclaration();
```

Comment:
this is not always true!

Reply:
The comment says you are unsure about this.
In this case don't make it an assert.

Assertion is defined as: a statement that you strongly believe is true.

Reply:
Yes, I am aware that the assert is not always true, but on the other hand I have no clue what other possible forms the CommaExp might have in this particular scenario, therefore, instead of letting a potential segfault with no additional information happen, I opted for an assert that gives you the exact line that is causing problems.